openedx/features/survey_report/context_processors.py

```diff
@@ -0,0 +1,31 @@
+"""
+This is the survey report contex_processor modules
+
+This is meant to determine the visibility of the survey report banner
+across all admin pages in case a survey report has not been generated
+
+"""
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta  # for months test
+from .models import SurveyReport
+
+
+def admin_extra_context(request):
+    """
+    This function sends extra context to every admin site
+
+    The current treshhold to show the banner is one month but this can be redefined in the future
+
+    """
+    months = 1
```

Comment:
We should move this to a setting so that it is configurable and also put a better default. Since we would like to get reports about twice a year, we could set the default to 6.

Reply:
Thanks for the feedback ioll change it to 6